src/VisualStudio/Core/Test/ProjectSystemShim/VisualStudioProjectTests/AnalyzerReferenceTests.vb

```diff
@@ -108,6 +111,100 @@ Namespace Microsoft.VisualStudio.LanguageServices.UnitTests.ProjectSystemShim
             End Using
         End Function
 
+        <WpfFact>
+        Public Async Function RazorSourceGenerator() As Task
+            Using environment = New TestEnvironment()
+                Dim providerFactory = DirectCast(environment.ExportProvider.GetExportedValue(Of IVisualStudioDiagnosticAnalyzerProviderFactory), MockVisualStudioDiagnosticAnalyzerProviderFactory)
+                providerFactory.Extensions =
+                {
+                    ({
+                        Path.Combine(TempRoot.Root, "RazorVsix", "Microsoft.NET.Sdk.Razor.SourceGenerators.dll"),
+                        Path.Combine(TempRoot.Root, "RazorVsix", "VsixDependency1.dll"),
+                        Path.Combine(TempRoot.Root, "RazorVsix", "VsixDependency2.dll")
+                     },
+                     "Microsoft.VisualStudio.RazorExtension"),
+                     ({
+                        Path.Combine(TempRoot.Root, "File.dll")
+                     },
+                     "AnotherExtension")
+                }
+
+                Dim project = Await environment.ProjectFactory.CreateAndAddToWorkspaceAsync(
+                    "Project", LanguageNames.CSharp, CancellationToken.None)
+
+                ' adding just Razor dependency and not the main source generator is a no-op
+                Using project.CreateBatchScope()
```

Comment:
You don't need the batch scopes in this case; not hurting anything but would tighten up the test code.